src/debugger/scriptImporter.ts

```diff
@@ -71,7 +71,7 @@ export class ScriptImporter {
      * Writes the script file to the project temporary location.
      */
     private writeAppScript(scriptBody: string, scriptUrl: url.Url): Q.Promise<String> {
-        let scriptFilePath = path.join(this.sourcesStoragePath, scriptUrl.pathname); // scriptFilePath = "$TMPDIR/index.ios.bundle"
+        let scriptFilePath = path.join(this.sourcesStoragePath, path.basename(scriptUrl.pathname)); // scriptFilePath = "$TMPDIR/index.ios.bundle"
```

Comment:
The source files refer to a sourcemap at the end of the file, and we do some tweaks to that (see https://github.com/Microsoft/vscode-react-native/blob/master/src/debugger/sourceMap.ts#L70 ) to make sure that it refers to the correct file. As long as the `foo.js` file refers to a `bar.js.map` file in a sourcemap comment, I believe that the debugger should correctly pick up the source map.